src/main/webapp/src/app/recommendation-form/recommendation-form.component.spec.ts

```diff
@@ -0,0 +1,25 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { RecommendationFormComponent } from './recommendation-form.component';
+
+describe('RecommendationFormComponent', () => {
+  let component: RecommendationFormComponent;
+  let fixture: ComponentFixture<RecommendationFormComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [ RecommendationFormComponent ]
+    })
+    .compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(RecommendationFormComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+});
```

Comment:
Can you add a test that fetches the html element for this component and then sets all the user inputs in the form, and then check preferencesForm inside the component class has been updated? (feel free to message me if you're not sure how to do this)

Reply:
done?